lib/pure/os.nim

```diff
@@ -972,6 +972,14 @@ proc rawCreateDir(dir: string): bool =
       result = false
     else:
       raiseOSError(osLastError(), dir)
+  elif defined(haiku):
+    let res = mkdir(dir, 0o777)
+    if res == 0'i32:
+      result = true
+    elif errno in {EEXIST, EROFS}:
```

Comment:
@Araq 

For some reasons `chckRange` is generated here, with range from 0 to 65535. This obviously broke on Haiku because we use some weird negative numbers. Do you happen to know where I should look at?

Reply:
Try this variant:

```nim
elif errno in [EEXIST, EROFS]:
```

Reply:
It worked, but the generated C result in an array passed to `contains()`. Can I trust that the C compiler will optimize this away or should I go for this variant

```nim
elif errno == EEXIST or errno == EROFS:
```

?

Reply:
hmm better use your ``or`` version.